### PR TITLE
Fix lambda (fixes #859)

### LINKED ISSUE #859: setting lambda in a TRR does not work anymore
### Expected behaviour
I want to create a TRR from a DCD while also setting the lambda (which used to be `ts.lmbda`), for instance
```python
import izip
import numpy as np
import MDAnalysis   
L = MDAnalysis.Universe("./rotamer1_R1A_298K.pdb", "./rotamer1_R1A_298K.dcd")
pop = np.loadtxt("R1A_298K_populations.dat")
with MDAnalysis.Writer(trr, L.atoms.n_atoms) as W:
        for ts, p in izip(L.trajectory, pop):
            ts.lmbda = p  # add lambda for TRR
            W.write(ts)

# then test
def test(trr="rotamer1_R1A_298K.trr"):
    u = MDAnalysis.Universe("rotamer1_R1A_298K.pdb", trr)
    pp = np.array([ts.lmbda for ts in u.trajectory])
    ref = np.loadtxt("R1A_298K_populations.dat")
    assert np.abs(pp - ref).max() < 1e-10

test()
```

### Actual behaviour
The newly written TRR either does not expose lambda or it wasn't written:

```
---> 41     pp = np.array([ts.lmbda for ts in u.trajectory])

AttributeError: 'Timestep' object has no attribute 'lmbda'

ipdb> ts.data
{'step': 0, 'dt': 1.0, 'time': 0.0}
```

### Currently version of MDAnalysis:

0.15.1-dev0

(run `python -c "import MDAnalysis as mda; print(mda.__version__)"`)

### PR DESCRIPTION
**Pull request by kain88-de**
Fixes #859

Changes made in this Pull Request:
 - lambda values are saved in the data dictionary of a TimeStep

PR Checklist
------------
 - [x] Tests?
 - [x] Docs?
 - [x] CHANGELOG updated?
 - [x] Issue raised/referenced?

### CODE REVIEW

File: package/MDAnalysis/coordinates/TRR.py [hunk 1]
```diff
@@ -79,7 +84,12 @@ def write_next_timestep(self, ts):
 
         box = triclinic_vectors(dimensions)
 
-        self._xdr.write(xyz, velo, forces, box, step, time, 1, self.n_atoms)
+        lmbda = 1
```
Comment (richardjgowers) [inline package/MDAnalysis/coordinates/TRR.py]: Isn't it better to one line this with get?

File: package/MDAnalysis/lib/formats/libmdaxdr.pyx [hunk 2]
```diff
@@ -423,7 +423,7 @@ cdef class TRRFile(_XDRFile):
         return_code = 1
         cdef int step = 0
         cdef int has_prop = 0
-        cdef float time = 0
+        cdef float time = 1
```
Comment (richardjgowers) [inline package/MDAnalysis/lib/formats/libmdaxdr.pyx]: Was this an issue before?

Comment (kain88-de) [inline package/MDAnalysis/lib/formats/libmdaxdr.pyx]: No but I wanted to be consistent.

File: package/MDAnalysis/coordinates/TRR.py [hunk 3]
```diff
@@ -24,6 +24,10 @@ class TRRWriter(XDRBaseWriter):
     used by other Gromacs tools to store and process other data such as modes
     from a principal component analysis.
 
+    If the data dictionary of a TimeStep contains the key 'lambda' the
+    corresponding value will be used as the lambda value for written TRR file.
+    If None is found the lambda is set to 1.
```
Comment (orbeckst) [inline package/MDAnalysis/coordinates/TRR.py]: In Gromacs, `lambda == 0` typically means "full interaction" so a default of 0 would be better.

This is also the default in Gromacs TRRs:
```
md.trr frame 0:
   natoms=    140246  step=         0  time=0.0000000e+00  lambda=         0
```

File: package/MDAnalysis/lib/formats/libmdaxdr.pyx (hunk 2 above)
Comment (orbeckst) [inline package/MDAnalysis/lib/formats/libmdaxdr.pyx]: Default for lambda should be 0 (see comment above).

File: package/MDAnalysis/lib/formats/libmdaxdr.pyx [hunk 4]
```diff
@@ -423,7 +423,7 @@ cdef class TRRFile(_XDRFile):
...
         cdef float lmbda = 0
```
Comment (orbeckst) [inline package/MDAnalysis/lib/formats/libmdaxdr.pyx]: Default for lambda should be 0 (see comment above).

File: package/MDAnalysis/coordinates/TRR.py (hunk 3 above)
Comment (kain88-de) [inline package/MDAnalysis/coordinates/TRR.py]: 0 it is then

File: package/MDAnalysis/lib/formats/libmdaxdr.pyx (hunk 2 above)
Comment (kain88-de) [inline package/MDAnalysis/lib/formats/libmdaxdr.pyx]: and I was in the wrong long apparently.

**Comment by orbeckst** [comment]
Travis is failing after the merge but this looks like an intermittent failure. The first time one of the runs failed and the first error was related to *too many files open*. I then restarted (https://travis-ci.org/MDAnalysis/mdanalysis/jobs/132190530) and got another random(?) failure `IOError: [Errno 5] Cannot open file or stream in mode='r'.: "'/var/folders/gw/_2jq29095y7b__wtby9dg_5h0000gn/T/tmp8zFXapwrite-atoms-test.xyz.bz2'"`.

I am just hitting "redo" again but I think others had similar intermittent failures. It would be good to know what to do about it.

**Comment by kain88-de** [comment]
This happens on osx right?

**Comment by orbeckst** [comment]
Correct, happened on OSX.
And the rerun just passed.

**Comment by richardjgowers** [comment]
I'm not sure I've ever seen it on linux, just osx

**Pull request merged**